src/Authentication/AzureArtifacts.cs

```diff
@@ -51,7 +51,12 @@ public static PublicClientApplicationBuilder WithBroker(this PublicClientApplica
                     ListOperatingSystemAccounts = true,
                     MsaPassthrough = true
                 })
-            .WithParentActivityOrWindow(() => GetConsoleOrTerminalWindow());
+            .WithParentActivityOrWindow(() => parentWindowHandle ?? GetConsoleOrTerminalWindow());
+    }
```

Comment:
@TedyHH - this has no debug logging on the passed in parent windows handler - would your scenario benefit from any debug logs here? 

Otherwise, this PR works on a simple windows form app.